timescaledb/quick-start/python.md

```diff
@@ -226,73 +221,100 @@ Congratulations, you've successfully inserted data into TimescaleDB using Python
+                               %s as sensor_id,
+                               random()*100 AS temperature,
+                               random() AS cpu
+                            """
+    cursor.execute(simulate_query, data)
+    values = cursor.fetchall()
+```
+
 ### Step 2: Define columns of table you're inserting data into
 Then we define the column names of the table we want to insert data into. In this
 case, we're using the `sensor_data` hypertable that we created in the
 "Generate a Hypertable" section above. This hypertable consists of the columns
 named `time`, `sensor_id`, `temperature` and `cpu`. We define these column names
-in a tuple of strings called `cols`.
+in a list of strings called `cols`.
+
+```python
+cols = ['time', 'sensor_id', 'temperature', 'cpu']
+```
 
 ### Step 3: Instantiate a CopyManager with your target table and column definition
 Lastly we create an instance of the `pgcopy` CopyManager, `mgr`, and pass our
-connection variable, hypertable name, and tuple of column names. Then we use
+connection variable, hypertable name, and list of column names. Then we use
 the `copy` function of the CopyManager to insert the data into the database
-performantly using `pgcopy` and then commit when we're done. There is also sample
-code to check if the insert worked.
+performantly using `pgcopy`.
+
+```python
+mgr = CopyManager(conn, 'sensor_data', cols)
+mgr.copy(values)
+```
+
+Finally, commit to persist changes:
+
+```python
+conn.commit()
+```
+
+Full sample code to insert data into TimescaleDB using `pgcopy`, using the example of sensor data from four sensors:
+
+```python
+# insert using pgcopy
+def fast_insert(conn):
+    cursor = conn.cursor()
+
+    # for sensors with ids 1-4
+    for id in range(1, 4, 1):
+        data = (id,)
+        # create random data
+        simulate_query = """SELECT generate_series(now() - interval '24 hour', now(), interval '5 minute') AS time,
+                           %s as sensor_id,
+                           random()*100 AS temperature,
+                           random() AS cpu
+                        """
+        cursor.execute(simulate_query, data)
+        values = cursor.fetchall()
+
+        # column names of the table you're inserting into
+        cols = ['time', 'sensor_id', 'temperature', 'cpu']
+
+        # create copy manager with the target table and insert
+        mgr = CopyManager(conn, 'sensor_data', cols)
+        mgr.copy(values)
+
+    # commit after all sensor data is inserted
+    # could also commit after each sensor insert is done
+    conn.commit()
+```
+
+You can also check if the insertion worked:
+
+```python
+cursor.execute("SELECT * FROM sensor_data LIMIT 5;")
+print(cursor.fetchall())
+```
 
 Congratulations, you've successfully performantly inserted data into TimescaleDB
 using Python and the `pgcopy` library.
 
-
 ## Execute a query [](execute_query)
```

Comment:
So, totally NOT your fault, but it's weird that this tutorial jumps to a different dataset. And, now I'm realizing that all of these tutorials are a little bit different which is annoying and we'll work to standardize at some level.

Let's make this section about querying the data from what was just created above... the `sensor_data` table. Use the Golang tutorial to update it.

Reply:
I really really want to re-write all of these tutorials 😬

Reply:
Yeah, totally agree. I think we're close. Once we have one decent use case (like the prototypical "To-do app"), I'd like to give all of these a pass... and then we can work on the next example app, etc. Each should have something that's unique to help demonstrate features/uses.

Reply:
https://github.com/timescale/docs/commit/00ddcad84ce240d25d2f72b65148b8cb1ea13461